skimmer/skimmer_utils.py

```diff
@@ -39,11 +79,21 @@ def apply_trigger_cut(events, trigger_list):
         ak.Array
     """
 
-    trigger_filter = ak.zeros_like(events.EvtNum, dtype=bool)
-    for trigger_name in trigger_list:
-        trigger_index = trigger_table[trigger_name]
-        trigger_branch = events.TriggerPass[:, trigger_index]
-        trigger_filter = trigger_filter | (trigger_branch == 1)
+    #check if TriggerPass is inside events or not, and use it if it is (for treemaker skims)
+    if "TriggerPass" in events.fields:
+        trigger_filter = ak.zeros_like(events.EvtNum, dtype=bool)
+        for trigger_name in trigger_list:
+            trigger_index = trigger_table_treemaker[trigger_name]
+            trigger_branch = events.TriggerPass[:, trigger_index]
+            trigger_filter = trigger_filter | (trigger_branch == 1)
+    else: #if TriggerPass is not inside events, nano trigger skim
```

Comment:
The triggers in NanoAOD are not stored as an array, I don't see why you need the `trigger_table_nanoaod`

Reply:
solved